csrc/ec_gen.cpp

```diff
@@ -183,16 +186,17 @@ JNIEXPORT void JNICALL Java_com_amazon_corretto_crypto_provider_EcGen_generateEc
         // Now that we have the params, extract the group from them (all we care about)
         CHECK_OPENSSL(group = EC_KEY_get0_group(ecParams));
 
-        // Build the structure which will hold our result
-        if(!likely(key = EC_KEY_new())) {
-            throw java_ex(EX_OOM, "Out of memory");
-        }
-
-        CHECK_OPENSSL(EC_KEY_set_group(key, group));
+        // Setup the temporary PKEY context with the extracted parameters.
+        CHECK_OPENSSL(tmpCtx = EVP_PKEY_CTX_new_id(EVP_PKEY_EC, NULL));
+        CHECK_OPENSSL(EVP_PKEY_paramgen_init(tmpCtx) == 1);
+        CHECK_OPENSSL(EVP_PKEY_CTX_set_ec_paramgen_curve_nid(tmpCtx, EC_GROUP_get_curve_name(group)) == 1);
+        CHECK_OPENSSL(EVP_PKEY_paramgen(tmpCtx, &pkey) == 1);
 
-        if (!likely(EC_KEY_generate_key(key))) {
-            throw_openssl("Unable to generate key");
-        }
+        // Generate the actual EC key.
+        CHECK_OPENSSL(keyCtx = EVP_PKEY_CTX_new(pkey, NULL));
+        CHECK_OPENSSL(EVP_PKEY_keygen_init(keyCtx) == 1);
+        CHECK_OPENSSL(EVP_PKEY_keygen(keyCtx, &pkey) == 1);
+        CHECK_OPENSSL(key = EVP_PKEY_get1_EC_KEY(pkey));
```

Comment:
Can we switch to `EVP_PKEY_get0_EC_KEY()` which removes the need to free the returned `EC_KEY` object? (That API was introduced after ACCP launched on 1.0.x, so wasn't available originally.)